### PR TITLE
Fix a bug with q when tsen is used as control variable in fv3 regional EnKF analysis

### PR DESCRIPTION
**Pull request by BinLiu-NOAA**
Fix a bug with q when tsen is used as control variable in fv3 regional EnKF analysis (From @TingLei-NOAA)

### CODE REVIEW

File: src/enkf/gridio_fv3reg.f90 [hunk 1]
```diff
@@ -361,12 +362,12 @@ subroutine readgriddata(nanal1,nanal2,vars3d,vars2d,n3d,n2d,levels,ndim,ntimes,f
 end subroutine readgriddata
 
   !========================================================================
-  ! readgriddata_nmm.f90: read FV3-Lam state or control vector
+  ! readgriddata_nmm.f90: read WRF-NMM state or control vector
```
Comment (BinLiu-NOAA) [inline src/enkf/gridio_fv3reg.f90]: @TingLei-NOAA, looks like this change need to be reverted?

File: src/enkf/gridio_fv3reg.f90 [hunk 2]
```diff
@@ -361,12 +362,12 @@ subroutine readgriddata(nanal1,nanal2,vars3d,vars2d,n3d,n2d,levels,ndim,ntimes,f
...
   !-------------------------------------------------------------------------
 
 
   !========================================================================
-  ! writegriddata.f90: write FV3-LAM analysis
+  ! writegriddata.f90: write WRF-ARW or WRF-NMM analysis
```
Comment (BinLiu-NOAA) [inline src/enkf/gridio_fv3reg.f90]: @TingLei-NOAA, looks like this change need to be reverted?

File: src/enkf/gridio_fv3reg.f90 [hunk 3]
```diff
@@ -518,81 +519,86 @@ subroutine writegriddata(nanal1,nanal2,vars3d,vars2d,n3d,n2d,levels,ndim,vargrid
...
+       call write_fv3_restart_data3d(varstrname,fv3filename,file_id,qworkvar3d)
+       do k=1,nlevs
+          if (nproc .eq. 0)                                               &
+             write(6,*) 'WRITEregional : sphum ',                           &
+                 & k, minval(qworkvar3d(:,:,k)), maxval(qworkvar3d(:,:,k))
+       enddo
+    end if
+
+
+
```
Comment (BinLiu-NOAA) [inline src/enkf/gridio_fv3reg.f90:543]: Please help to fix the indentation in this code section: lines 522-547 if possible.

File: src/enkf/gridio_fv3reg.f90 [hunk 4]
```diff
@@ -518,81 +519,86 @@ subroutine writegriddata(nanal1,nanal2,vars3d,vars2d,n3d,n2d,levels,ndim,vargrid
...
+           do j=1,ny_res
+              do i=1,nx_res
+                 nn=nn+1
+                 workinc3d(i,j,k)=vargrid(nn,levels(tsen_ind-1)+k,nb,ne) 
+              enddo
+           enddo
+           enddo
+            call read_fv3_restart_data3d(varstrname,fv3filename,file_id,workvar3d)
+               workvar3d=workvar3d+workinc3d
+            call write_fv3_restart_data3d(varstrname,fv3filename,file_id,workvar3d)
```
Comment (BinLiu-NOAA) [inline src/enkf/gridio_fv3reg.f90]: Please help to fix the indentation in this code section if possible.

File: src/enkf/gridio_fv3reg.f90 [hunk 5]
```diff
@@ -518,81 +519,86 @@ subroutine writegriddata(nanal1,nanal2,vars3d,vars2d,n3d,n2d,levels,ndim,vargrid
...
-      enddo
-       qworkvar3d=qworkvar3d+workinc3d   
-       endif
+        call read_fv3_restart_data3d(varstrname,fv3filename,file_id,qbgworkvar3d)
+      endif 
+       tvworkvar3d=tsenworkvar3d*(one+fv*qbgworkvar3d)
+       tvworkvar3d=tvworkvar3d+workinc3d
+      if(.not. ( q_ind > 0)) then
+       tsenworkvar3d=tvworkvar3d/(one+fv*qbgworkvar3d)
+      else
```
Comment (BinLiu-NOAA) [inline src/enkf/gridio_fv3reg.f90]: Please help to fix the indentation in this code section if possible.

**Review by BinLiu-NOAA — COMMENTED** [review]
@TingLei-NOAA Could you please fix the indentation formatting issues below if possible. Thanks!

**Comment by TingLei-NOAA** [comment]
Hi, Bin,
I created a new PR https://github.com/NOAA-EMC/GSI/pull/134.
Because  some messing up I made  when I commit --amend those formatting
changes to the previous branch/PR.
Hence , finally I created this new branch and new PR.
Please let me know if there are any issues.
Ting

______________________________

Ting Lei

IMSG at NOAA/NWS/NCEP/EMC

5830 University Research Ct., Cubicle 2765

College Park, MD 20740

***@***.*** ***@***.***>

301-683-3624

**Comment by BinLiu-NOAA** [comment]
Thank you very much, @TingLei-NOAA! And deleting the feature/esnda-bugfix_q branch now.

**Pull request merged**